desktop-widgets/mainwindow.cpp

```diff
@@ -137,14 +137,14 @@ MainWindow::MainWindow() : QMainWindow(),
 	// widgets will change on the mainwindow.
 
 	// for the "default" mode
-	MainTab *mainTab = new MainTab();
-	DiveListView *diveListView = new DiveListView();
-	ProfileWidget2 *profileWidget = new ProfileWidget2();
+	information = new MainTab(this);
```

Comment:
what's wrong with calling this mainTab? I feel a bit uneasy with these automatically typed variables but maybe only because I am not used to these and do not immediately spot those as declarations.

Reply:
`mainTab` would be the right name indeed. I just didn't want to change too much in this commit and for some (historical?) reason this was named `information` before.

> I feel a bit uneasy with these automatically typed variables but maybe only because I am not used to these and do not immediately spot those as declarations.

Not sure what you mean here - the declaration is in the `mainwindow.h` file. I was thinking about moving these initializations into the initializer list:
```
MainWindow::MainWindow() :
...
	mainTab(new MainTab(this)),
	diveListView(new diveListView(this)),
...
```
Perhaps this would make it more explicit that these are member-variables?

Reply:
I pushed the variable renaming as a second commit. But probably this should be squashed into the first.

Then I noted that actually we can't initialize the sub-widgets in the initializer-list, because the constructors of the widgets use `MainWindow::instance()`, which is not yet set! To me the whole thing looks very brittle or even undefined-behavior-ish: The widgets happily use `MainWindow` during construction of the latter.